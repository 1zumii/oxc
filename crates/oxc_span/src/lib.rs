//! Source positions and related helper functions.
//!
//! <https://doc.rust-lang.org/beta/nightly-rustc/rustc_span>

mod atom;
mod compact_str;
mod source_type;
mod span;

pub mod cmp;
pub mod hash;

pub use crate::{
    atom::Atom,
    compact_str::{CompactStr, MAX_INLINE_LEN as ATOM_MAX_INLINE_LEN},
    source_type::{
        Language, LanguageVariant, ModuleKind, SourceType, UnknownExtension, VALID_EXTENSIONS,
    },
    span::{GetSpan, GetSpanMut, Span, SPAN},
};
