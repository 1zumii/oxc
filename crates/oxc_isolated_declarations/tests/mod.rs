use std::{fs, path::Path};

use oxc_allocator::Allocator;
use oxc_codegen::CodeGenerator;
use oxc_parser::Parser;
use oxc_span::SourceType;

fn transform(path: &Path, source_text: &str) -> String {
    let allocator = Allocator::default();
    let source_type = SourceType::from_path(path).unwrap();
    let program = Parser::new(&allocator, source_text, source_type).parse().program;
    CodeGenerator::new().build(&program).source_text
}

#[test]
fn snapshots() {
    insta::glob!("fixtures/*.{ts,tsx}", |path| {
        let source_text = fs::read_to_string(path).unwrap();
        let snapshot = transform(path, &source_text);
        let name = path.file_stem().unwrap().to_str().unwrap();
        insta::with_settings!({ prepend_module_to_snapshot => false, snapshot_suffix => "", omit_expression => true }, {
            insta::assert_snapshot!(name, snapshot);
        });
    });
}
