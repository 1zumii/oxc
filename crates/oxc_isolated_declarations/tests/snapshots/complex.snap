---
source: crates/oxc_isolated_declarations/tests/mod.rs
input_file: crates/oxc_isolated_declarations/tests/fixtures/complex.ts
---
class B {}
